tests/integration/device_test.sh

```diff
@@ -4,87 +4,223 @@ include './tests/unit/utils.sh'
 include './tests/integration/utils.sh'
 include './src/device_info.sh'
 
+declare -gA DEVICE_INFO_RESULTS
+
 function oneTimeSetUp()
 {
   # The VERBOSE variable is set and exported in the run_tests.sh script based
   # on the command-line options provided by the user. It controls the verbosity
   # of the output during the test runs.
   setup_container_environment "$VERBOSE" 'device'
+  # Ensure kw_device_output is called once to populate the results array
+  kw_device_output
```

Comment:
There is no need to create this function since you just call it once. Just move the content of this function to the oneTimeSetUp.